quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/Picocli.java

```diff
@@ -558,124 +551,62 @@ private static void outputDisabledProperties(Set<String> properties, boolean bui
-        Optional<String> currentProfile = ofNullable(org.keycloak.common.util.Environment.getProfile());
-        Optional<String> persistedProfile = getBuildTimeProperty("kc.profile");
-
-        if (!persistedProfile.orElse("").equals(currentProfile.orElse(""))) {
-            return true;
-        }
-
-        for (String propertyName : getConfig().getPropertyNames()) {
-            // only check keycloak build-time properties
-            if (!isBuildTimeProperty(propertyName)) {
-                continue;
-            }
-
-            ConfigValue configValue = getConfig().getConfigValue(propertyName);
-
-            if (configValue == null || configValue.getConfigSourceName() == null) {
-                continue;
-            }
-
-            // try to resolve any property set using profiles
-            if (propertyName.startsWith("%")) {
-                propertyName = propertyName.substring(propertyName.indexOf('.') + 1);
-            }
-
-            String persistedValue = getBuildTimeProperty(propertyName).orElse("");
-            String runtimeValue = getRuntimeProperty(propertyName).orElse(null);
-
-            // compare only the relevant options for this command, as not all options might be set for this command
-            if (cmdCommand.getCommand() instanceof AbstractCommand) {
-                AbstractCommand abstractCommand = cmdCommand.getCommand();
-                PropertyMapper<?> mapper = PropertyMappers.getMapper(propertyName);
-                if (mapper != null) {
-                    if (!abstractCommand.getOptionCategories().contains(mapper.getCategory())) {
-                        continue;
-                    }
+    public static Properties getNonPersistedBuildTimeOptions() {
+        Properties properties = new Properties();
+        // TODO: could get only non-persistent property names
+        Configuration.getPropertyNames().forEach(name -> {
+            boolean quarkus = false;
+            PropertyMapper<?> mapper = PropertyMappers.getMapper(name);
+            if (mapper != null) {
+                if (!mapper.isBuildTime()) {
+                    return;
                 }
-            }
-
-            if (runtimeValue == null && isNotBlank(persistedValue)) {
-                PropertyMapper<?> mapper = PropertyMappers.getMapper(propertyName);
-
-                if (mapper != null && persistedValue.equals(Option.getDefaultValueString(mapper.getDefaultValue().orElse(null)))) {
-                    // same as default
-                    continue;
+                name = mapper.getFrom();
+                if (properties.containsKey(name)) {
+                    return;
                 }
-
-                // probably because it was unset
-                return true;
+            } else if (name.startsWith(MicroProfileConfigProvider.NS_QUARKUS)) {
+                // TODO: this is not correct - we are including runtime properties here, but at least they 
```

Comment:
Quarkus needs to have some mechanism for testing whether it's buildtime option, or not, right?

Reply:
Yes, for most of the configuration in quarkus it is able to determine if something is build time vs. runtime and tries to capture in the respective bucket - runtime defaults, build time values, etc. If we worked with the quarkus team more we could probably come up with a way to just rely on their internal stuff, rather than needing our own persisted properties.